Tools/ardupilotwaf/toolchain.py

```diff
@@ -24,14 +25,71 @@ def configure(cfg):
     OBJCOPY='objcopy',
 )
 
+def find_realexec_path(cfg, filename, path_list=[]):
+    if not filename:
+        return ''
+
+    if not path_list:
+        path_list = cfg.environ.get('PATH','').split(os.pathsep)
+
+    for dir in path_list:
+        path = os.path.abspath(os.path.expanduser(os.path.join(dir, filename)))
+
+        if os.path.isfile(path):
+            if os.path.islink(path):
+                realpath = os.path.realpath(path)
+
+                if filename not in os.path.basename(realpath):
+                    continue
+                else:
+                    return os.path.dirname(realpath)
+
+            else:
+                return os.path.dirname(path)
+
+    cfg.fatal('Could not find real exec path to %s in path_list %s:' % (filename, path_list))
+
 def configure(cfg):
-    if not cfg.env.TOOLCHAIN:
-        cfg.env.TOOLCHAIN = 'native'
-        prefix = ''
-    else:
-        cfg.env.TOOLCHAIN = Utils.to_list(cfg.env.TOOLCHAIN)[0]
-        cfg.msg('Using toolchain prefix', cfg.env.TOOLCHAIN)
-        prefix = cfg.env.TOOLCHAIN + '-'
-
-    for k in suffixes:
-        cfg.env.append_value(k, prefix + suffixes[k])
+    toolchain = boards.get_board(cfg.env.BOARD).toolchain
+
+    if toolchain != 'native':
+        cfg.msg('Using toolchain prefix', toolchain)
+        prefix = toolchain + '-'
+
+        c_compiler = cfg.options.check_c_compiler or 'gcc'
+        cxx_compiler = cfg.options.check_cxx_compiler or 'g++'
```

Comment:
That doesn't work in a non-native build. Clang always needs the GCC toolchain for a non-native build.